### PR TITLE
CLN: remove unused entries from _lite_rule_alias

### PR DESCRIPTION
**Pull request by natmokval**
xref #56346

remove unused entries `"min", "ms", "us", "ns"` from _lite_rule_alias

### CODE REVIEW

File: pandas/_libs/tslibs/offsets.pyx [hunk 1]
```diff
@@ -4672,6 +4668,9 @@ _dont_uppercase = {
     "qe-oct",
     "qe-nov",
     "ye",
+    "min",
+    "ns",
+    "us",
```
Comment (MarcoGorelli) [inline pandas/_libs/tslibs/offsets.pyx:4673]: can you remove unrelated changes please?

**Review by MarcoGorelli — COMMENTED** [review]
thanks, should be good, just a comment

File: pandas/_libs/tslibs/offsets.pyx (hunk 1 above)
Comment (natmokval) [reply pandas/_libs/tslibs/offsets.pyx:4673]: If we remove these from _`dont_uppercase` the tests `test_offset_freqstr()` and `test_rule_code()` in `pandas/tests/tseries/offsets/test_offsets.py` will fail. The reason: we uppercase names of offsets class in `_get_offset()`.  In PR #56346 where we deprecate uppercasing/lowercasing we will remove them from `_dont_uppercase`

Comment (MarcoGorelli) [reply pandas/_libs/tslibs/offsets.pyx:4673]: ah, thanks - so, they're not currently unused then? but will become unused once https://github.com/pandas-dev/pandas/pull/56346 is enforced? should this PR wait until after that one then?

Comment (natmokval) [reply pandas/_libs/tslibs/offsets.pyx:4673]: I checked in PR #56346, we should keep `"min", "us", "ns"` in the list _dont_uppercase to pass these tests.
On the other hand, we have all lowercase offsets frequencies in this list (`'h', 's'`, etc.), and it looks more clear then have these entries (`"min": "min", "us": "us"`, etc.) in dict `_lite_rule_alias`.

Comment (MarcoGorelli) [reply pandas/_libs/tslibs/offsets.pyx:4673]: thanks for explaining

sorry for not having got round to this yet, was quite this week with a client project - but I'll get to it, thanks for looking at it all so carefully!

Comment (natmokval) [reply pandas/_libs/tslibs/offsets.pyx:4673]: thank you for helping me with my PRs!

File: pandas/tests/indexes/datetimes/test_partial_slicing.py [hunk 2]
```diff
@@ -236,7 +236,7 @@ def test_partial_slice_second_precision(self):
         rng = date_range(
             start=datetime(2005, 1, 1, 0, 0, 59, microsecond=999990),
             periods=20,
-            freq="US",
+            freq="us",
```
Comment (MarcoGorelli) [inline pandas/tests/indexes/datetimes/test_partial_slicing.py:239]: does this PR make `to_offset('US')` suddently start failing? if so, I don't think we can do this yet (at least, not before https://github.com/pandas-dev/pandas/pull/56346/files ?)

Comment (natmokval) [reply pandas/tests/indexes/datetimes/test_partial_slicing.py:239]: > does this PR make `to_offset('US')` suddently start failing? if so, I don't think we can do this yet (at least, not before https://github.com/pandas-dev/pandas/pull/56346/files ?)

yes, this test didn't fail for uppercase frequency `"US"` before this PR.

Comment (natmokval) [reply pandas/tests/indexes/datetimes/test_partial_slicing.py:239]: a small correction to the comment above: 
we fixed the `test_partial_slice_second_precision` in PR  DEPR: lowercase strings denoting freq for Week, MonthBegin, MonthEnd, etc. #56346 
but so far on main we have `to_offset('US')` in this test and it pass.

**Comment by MarcoGorelli** [comment]
I tried to do something similar in https://github.com/pandas-dev/pandas/pull/59240 - fancy taking a look please?

**Comment by natmokval** [comment]
> I tried to do something similar in #59240 - fancy taking a look please?
yeah, sure, I'll take a look.

**Comment by natmokval** [comment]
I got the same idea, to move  aliases "us", and"ns" from` _lite_rule_alias` to `c_PERIOD_AND_OFFSET_DEPR_FREQSTR`. 

seems like all tests passes except some tests in `pandas/tests/tseries/offsets/test_offsets.py`.

**Comment by natmokval** [comment]
closed as completed via #59240
